test/Microsoft.AspNetCore.Razor.Language.Test/Legacy/CSharpSectionTest.cs

```diff
@@ -42,11 +38,10 @@ public void ParseSectionBlockCapturesWhitespaceToEndOfLineInSectionStatementMiss
             // Arrange
             var chunkGenerator = new DirectiveChunkGenerator(SectionDirective.Directive);
             chunkGenerator.Diagnostics.Add(
-                RazorDiagnostic.Create(
-                    new RazorError(
-                        LegacyResources.FormatUnexpectedEOFAfterDirective(SectionDirective.Directive.Directive, "{"),
-                        new SourceLocation(25 + Environment.NewLine.Length, 0, 25 + Environment.NewLine.Length),
-                        length: 1)));
+                RazorDiagnosticFactory.CreateParsing_UnexpectedEOFAfterDirective(
+                    new SourceSpan(new SourceLocation(25 + Environment.NewLine.Length, 1, 4), contentLength: 1),
```

Comment:
Corrected the expectation here because we weren't comparing SourceLocation properly. I'll send a separate PR to fix that and the other related tests